### PR TITLE
Do instance delete in parallel for some fixtures in conftest

### PR DESCRIPTION
**Pull request by shyRozen**
Some teardown of multiple resource can take long time. If there are multiple resource of the same kind it will be deleted in parallel with threads.

Signed-off-by: Shay Rozen <shay.rozen@gmail.com>

### CODE REVIEW

**Review by ebenahar — DISMISSED** [review]

**Comment by shyRozen** [comment]
Removed a fix I did for some pv encryption test. Was not related to new finalizers functionality.

File: tests/conftest.py [hunk 1]
```diff
@@ -952,28 +955,24 @@ def finalizer():
...
+        def operation_delete(instance):
             if not instance.is_deleted:
-                pv_objs.append(instance.backed_pv_obj)
+                pv_obj = instance.backed_pv_obj
                 instance.delete()
                 instance.ocp.wait_for_delete(instance.name)
+                if (
+                    pv_obj.data.get("spec").get("persistentVolumeReclaimPolicy")
+                    == constants.RECLAIM_POLICY_RETAIN
+                    and pv_obj is not None
```
Comment (AviadP) [inline tests/conftest.py:967]:
```suggestion
                    and pv_obj 
```

Comment (shyRozen) [reply tests/conftest.py:967]: To tell you the truth, I just want to take the code as is and first make it threaded. I have another PR which I did the same and also changed the original code and I got too much work in debugging it. I just want to parallel the existing code right now without any adventures.

**Comment by ebenahar** [comment]
@shyRozen I think worth keeping this open and continue with this change whenever possible

**Pull request closed without merging**